pandas/tests/io/parser/compression.py

```diff
@@ -166,6 +166,21 @@ def test_read_csv_infer_compression(self):
 
         inputs[3].close()
 
+    def test_read_csv_infer_compression_pathlib(self):
+        """
+        Test that compression is inferred from pathlib.Path paths.
+        """
+        try:
```

Comment:
@TomAugspurger see https://github.com/pandas-dev/pandas/pull/17206/commits/b4e280f451cfd1f02954fd02667a7bc9d3271263. The fix should be simple I think, just passing `filepath_or_buffer` through the `_stringify_path` function in `_infer_compression`.